activerecord/lib/active_record/inheritance.rb

```diff
@@ -40,6 +40,8 @@ module Inheritance
       # This is true, by default.
       class_attribute :store_full_sti_class, instance_writer: false
       self.store_full_sti_class = true
+
+      @_type_candidates_cache = Concurrent::Map.new
```

Comment:
Definitely shouldn't be writing code that is just for tests to pass - you should be able to fix those with anonymous classes that inherit from `ActiveRecord::Base`.